### ISSUE TITLE
[Suggestion] Rework Rain scale bar and colors in the Maps tab

### ISSUE THREAD

**Issue by adem4ik** [FelixdelasPozas/TrayWeather#123]
Version: 1.23.0

- Open Maps
- Switch to Rain

**Result:** Rain is barely visible on the map, especially 0-6 mm. The problem comes from the gray colors on the gray/white background.

**Proposed result:** Use similar scale bar and colors as on https://openweathermap.org/weathermap?basemap=map&cities=false&layer=radar&lat=30&lon=-20&zoom=5

**Current scale bar:**
0.1 2 6 8 10 14 16 20 26 32 42 48 52 70+
**Proposed scale bar:**
0.1 0.5 1 2 4 6 8 10 12 14 16 24 32 60+

![image](https://user-images.githubusercontent.com/4707112/178101386-9fe078b6-c56d-4a81-8703-35550ae9b8bc.png)

**Comment by FelixdelasPozas** [comment]
Hi @adem4ik.

Yes, that will be much more visible for the rain map. I'll change it for the next release. I was also thinking on modifying also the temperature map scale because here in Spain we're going to be 40+ ºC next week and the maximum in the map is 36. So all the map will be in red... I'm thinking in making the scale bars dynamic depending on the current weather conditions for better "reading" the maps. I'll tinker with it for a bit to see if it works.

Regards, 
Félix.

**Comment by FelixdelasPozas** [comment]
Hello @adem4ik .

I've been playing a little bit with this and it's perfectly possible to do it, but only in the "advanced weather maps" of OpenWeatherMap. Those ["Maps 2.0"](https://openweathermap.org/api/weather-map-2), as they call it, allows the developer to specify a range for map values and associate a color to each. It looks good indeed but it [requires the developer subscription to OpenWeatherMaps (160€ a month)](https://openweathermap.org/price), and that's something I can't afford right now for a free project. For now I'll use the [free maps](https://openweathermap.org/api/weathermaps).

I'll leave the issue open so it remains visible, as this seems to be a frequently asked thing. 

Regards, 
Félix.

**Comment by alystair** [comment]
You could reach out to OpenWeatherMaps to see if they have any discounting/benefits for open source projects.

Alternatively potential costs could be deferred to users that specifically want the advanced functionality (using their own API keys or something) - although that seems like unnecessary feature creep :)

**Comment by FelixdelasPozas** [comment]
@alystair Yes, I suppose the application could try first to use the 2.0 maps and if it gets a "unauthorized" message from the server go back to use the old 1.0 maps. 
I'm going to change the way the application works with "data providers" to add different weather info sites. I'll try to add it in then, or at least test it. 

Regards, 
Félix.

**Comment by JohnLGalt** [comment]
Just out of curiosity, while I understand the need to maybe adjust the scale a bit for the unusually high temps the world saw this year, does the Map 1.0 not already include color gradients and scales? Was your reason for changing the colors from default for a specific reason? Being in the USA, I only know of the different sites and apps I've used for weather here, but it seems that, at least here, scaling for precipitation always starts from the lightest of greens and progresses through green, then yellow, then orange, then red for the hardest of rain(and using hues based on blue for frozen precipitation).

I've always wondered why TW had set colors and not gradients, and used different colors than what I normally see everywhere else.

**Comment by FelixdelasPozas** [comment]
The reason was that the lightgray colors where barely visible in the precipitation and cloud maps, so a gradient like the one proposed by @adem4ik (using greens instead of gray for lower values of rain) is much more visible. 
But in 1.0 maps you can ask for the rain map layer and the server will give you the map in a fixed gradient/colors then the application superimposes the rain layer to the "streetmap" layer (so OPW only gives you "the clouds" painted with the color it has, fixed). 
To modify the gradient as @adem4ik suggested is only available in the 2.0 maps API when you can specify temperature ranges and colors in a parameter called "palette". 
You can see the difference in parameters between [1.0](https://openweathermap.org/api/weathermaps) and [2.0](https://openweathermap.org/api/weather-map-2) maps. The [legend colors](https://openweathermap.org/map_legend) (what 2.0 calls palette) are fixed for 1.0 maps, so TW uses those colors.

**Comment by JohnLGalt** [comment]
Makes sense. Thanks for the  response and explanation.

**Comment by FelixdelasPozas** [comment]
I am closing this because with the OWM 1.0 is not possible and from version 1.31 every provider manages its own maps page (if provided), so the colors depend on the layers provided.
